site/content/integrate/apps/api/_index.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "API"
+heading: "Mattermost Apps API Reference"
+description: "This reference describes the Mattermost Apps API"
+weight: 200
+---
+
+## App Structure
+
+Apps consist of just 3 things:
+- [manifest.json]({{< ref "manifest" >}}) file
+- [Call]({{< ref "call" >}}) responders, starting with `/bindings` (see [Interactivity]({{< ref "interactivity" >}})) and `/on_install` (see [Manifest]({{< ref "manifest" >}}))
+- Static assets, currently just icon files
+
+## Mattermost Authentication
+An app declares in its `Manifest` the `Locations` that it will be binding to, and `Permissions` it will require to operate. These are consented to by the system administrator when installing Mattermost apps interactively, in Mattermost.
+
+Each app when installed into Mattermost gets an OAuth2 cientID/secret (not yet used), and a Bot User Account with a personal access token. Each call may receive a combination of `bot_access_token`, `acting_user_access_token`, and `admin_access_token` as applicable. See [Authentication]({{< ref "authentication" >}}) for more.
+
+Apps can use OAuth2 to connect to 3rd party APIs. See [3rd party OAuth2]({{< ref
+"3rdparty-oauth2" >}}) for more.
+
+
+
+
+Each call request an app receives contains a `bot_access_token` in the `context`. 
+
+Additionaly, if the app was granted `act_as_user` permission, and the call's
+`expand` contained `acting_user_access_token=all`, the call receives
+`acting_user_access_token` in the request `context`.
+
+Similarly, if the app was granted `act_as_admin` permission, the acting user is
+a system administrator, and the call's `expand` contained
+`admin_access_token=all`, the call receives `admin_access_token` in the
+`context`.
+
+
+## Interactivty: Bindings and Forms
+
+- To interact with users an App must bind a call to a location in the Mattermost
+  user interface. These locations may be `/`-commands, toolbar and menu items,
+  and embedded in posts.
+- A binding may display a `[Form]({{< ref "interactivity" >}})`, or it may
+  invoke a `Call` that will return a `Form`. Forms allow to gather fields with
+  limited dynamic behavior (dynamically-populated selects, form refresh on field
+  changes). A Form invokes a `Call` when it is submitted. A `/`=command is just
+  another way of filling out a `Form`.
+
+
+## Using Mattermost APIs
+- REST
```

Comment:
Are these APIs to be handled/described in this order below?

Reply:
This is still todo, perhaps @justinegeffen can help. This page is basically a decorated TOC for the API section.